clinica/iotools/converters/nifd_to_bids/nifd_to_bids_cli.py

```diff
@@ -8,13 +10,13 @@
 @cli_param.clinical_data_directory
 @cli_param.bids_directory
 def cli(
-    dataset_directory: str,
-    clinical_data_directory: str,
-    bids_directory: str,
+    dataset_directory: PurePath,
```

Comment:
It'd be great to have consistency across converters. At the moment, some converters (like `ADNI_to_bids`) have string types for paths, others (like `ukb_to_bids`) have `PathLike`, and this one will have `PurePath`. WDYT?

Reply:
Agreed. We are in that weird sport where the historic path manipulation library (`os.path`) is focused on strings, whilst the modern `pathlib` alternative uses `PurePath`.

`os.PathLike` provides a protocol to bridge between both at the expense of internal casting (to `str` or `Path` / `PurePath`) to please static type checking.

So far, this PR focused on fixing warnings such as type alignment issues. Whether to standardize to `PurePath` or `PathLike` would require a subsequent PR touching all converters.